src/backoff.ts

```diff
@@ -0,0 +1,41 @@
+const MAX_BACKOFF = 1000;
+
+/**
+ * A function that returns the next appropriate delay (ms) to use when backing off
+ */
+export type GetBackoffFunction = () => number;
+
+/**
+ * Creates a function that returns an exponential backoff delay. On each subsequent call, the function applies
+ * some logic to calculate the next value for the delay. The logic works like this: take the initial number,
+ * and after passing the next threshold, increase it. After reaching the last iteration threshold, the function
+ * no longer increases the number. The value can never exceed MAX_BACKOFF.
+ * @param initial The initial value to wait.
+ * @returns a function that returns the next backoff delay
+ */
+export function createGetBackoff(initial = 50): GetBackoffFunction {
```

Comment:
starting at 50 is probably too aggressive here. Not many queries will finish in 50ms, other than something like print 42 if we're lucky.

Reply:
@bardhi This is exactly what our own UI does.  It uses 50ms 10x then bumps up to 1000ms every time.  It's a pretty lame algorithm...but I wanted to mirror what we do on our end.  I posted a thread about this in Slack recently, didn't get any bites.  I'll revive that thread.

In the meantime, I totally agree 50ms is crazy.  I'll change that to 200ms at least here.

Reply:
That's what our backend does when retrying IO locally. I believe our own UI checks every 2 seconds.